src/sql/src/statements/copy.rs

```diff
@@ -27,6 +28,7 @@ macro_rules! format_sorted_hashmap {
         let mut result = String::new();
         for key in sorted_keys {
             if let Some(val) = hashmap.get(key) {
+                redact_sql_secrets(val);
```

Comment:
This is not how it works. You may instead check if key equals (case-insensitively) to `access_key_id` or `secret_access_key`.

Take a look at `redact_sql_secrets` and `SQL_SECRET_PATTERNS` so you may know how the previous logic actually works.